doc/proposals/webhook-proposal.md

```diff
@@ -0,0 +1,137 @@
+## Webhook Support
+
+Implementation Owner: theishshah
+
+Status: Draft
+
+[Background](#Background)
+
+[Goals](#Goals)
+
+[Design overview](#Design_overview)
+
+[User facing usage](#User_facing_usage)
+
+[Observations and open questions](#Observations_and_open_questions)
+
+### Background
+
+The upcoming stable version of controller runtime has support for running a webhook server and having webhooks to mutate or validate pods. The mutation webhook can change various attributes of a pod, and the validation webhook can read pod attributes and allow/deny a pod to run based on this information. 
+
+### Goals
+
+The goal of this proposal is to add the abilty for users to quickly add a webhook server and only implement the validation/mutation logic for their webhooks. Ideally this will support per-API webhooks, however, the proposal is not there yet.
+
+### Design overview
+
+All of the necesary files and changes for the generated operator occur in the `cmd/manager/` directory.
+
+osdk-webhook.go will be responsible for containing the logic to start up a webhook server, including a `webhookServer` struct to contain the necesary information to start the server.
+
+The webhook initilization logic function will be a function which returns successfully _without_ starting the server if the user does not enable webhook support via the command line (planned command is `$ operator-sdk generate webhooks`, more detail is outlined in the next section). If a user chooses to generate webhooks, the osdk-webhook.go file will be overwritten with a new file to contain a server init function with full functionality. Additionally it will generate the server information struct populated with the deafult values. Controller runtime defaults for these values are `Port: 9876` and `CertDir: "/tmp/cert"`.
```

Comment:
```suggestion
The webhook initialization logic function will be a function which returns successfully _without_ starting the server if the user does not enable webhook support via the command line (planned command is `$ operator-sdk generate webhooks`, more detail is outlined in the next section). If a user chooses to generate webhooks, the osdk-webhook.go file will be overwritten with a new file to contain a server init function with full functionality. Additionally it will generate the server information struct populated with the default values. Controller runtime defaults for these values are `Port: 9876` and `CertDir: "/tmp/cert"`.
```